lib/auth.ts

```diff
@@ -58,18 +61,37 @@ export const authOptions: NextAuthOptions = {
   },
   callbacks: {
     jwt: async ({ token, user }) => {
-      if (user) {
-        token.user = user;
+      const userDb = await getUser("local-108", token.sub as string);
+      if (userDb) {
+        return {
+          ...token,
+          name: userDb.name,
+          email: userDb.email,
+          image: userDb.imageURL,
+          id: userDb.id,
+          role: userDb.role,
+        };
       }
-      return token;
+      return {
+        ...token,
+        id: user.id,
+      };
     },
     session: async ({ session, token }) => {
       session.user = {
         ...session.user,
-        // @ts-expect-error
-        id: token.sub,
-        // @ts-expect-error
-        username: token?.user?.username || token?.user?.gh_username,
+        image: token.image as string,
+        id: token.sub as string,
+        email: token.email,
+        role: (token.role as string) || "user",
+        username: token?.username || token?.gh_username,
+      } as {
+        id: string;
+        role: string;
+        name?: string | null | undefined;
+        email?: string | null | undefined;
+        image?: string | null | undefined;
+        username?: string | null | undefined;
```

Comment:
was this necessary to modify in our case?

I'll need some explanation

Reply:
you right, we dont need the | null | undefined, but we do need to retype for typing purposes